skymp5-server/cpp/localization_provider/LocalizationProvider.h

```diff
@@ -0,0 +1,36 @@
+#include <filesystem>
+#include <map>
+#include <string>
+#include <vector>
+
+class DirectoryEntry
+{
+public:
+  uint32_t stringId;
+  uint32_t offset;
+  uint32_t length;
+  std::string str;
+
+  DirectoryEntry();
+  DirectoryEntry(uint32_t stringId, uint32_t stringOffset);
+};
+
+class LocalizationProvider
+{
+  std::vector<DirectoryEntry*> ParseDirectoryEntries(std::vector<char> buffer);
+
+  std::map<uint32_t, std::string> ParseStrings(
+    std::vector<char> buffer, std::vector<DirectoryEntry*> entries);
+
+  std::map<uint32_t, std::string> ParseILDLStrings(
+    std::vector<char> buffer, std::vector<DirectoryEntry*> entries);
```

Comment:
Don't copy heavy containers. Pass them by a reference. Use const for data that shouldn't be modified.
```suggestion
    const std::vector<char> &buffer, std::vector<DirectoryEntry*> &entries);
```

Reply:
if i use "const" i get C2440 error when trying to convert buffer to uint32_t

Reply:
If data shouldn't be changed use `reinterpret_cast<const type>(variable);`